core/utils.go

```diff
@@ -301,3 +311,65 @@ func FileExists(path string) (bool, error) {
 
 	return false, err
 }
+
+func needsBom() bool {
+	// The reason why we do this is because notepad.exe on Windows determines the
+	// encoding of an "empty" text file by the locale, for example, GBK in China,
+	// while golang string only handles utf8 well. However, a text file with utf8
+	// BOM header is not considered "empty" on Windows, and the encoding will then
+	// be determined utf8 by notepad.exe, instead of GBK or other encodings.
+	// This could be enhanced in the future by doing this only when a non-utf8
+	// locale is in use, and possibly doing that for any OS, not just windows.
+	return runtime.GOOS == "windows"
+}
+
+// get body text with a text editor form env
```

Comment:
I think the function name is enough of a description to skip out on this comment. It is also not a public function to need a super detailed comment.